dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/VelocityUtilTest.java

```diff
@@ -157,5 +158,36 @@ public void test_unable_to_modify_userid_in_velocity_session() throws Exception{
         assertTrue(request.getSession().getAttribute(WebKeys.USER_ID) ==null);
     }
     
+    
+    /**
+     * https://github.com/dotCMS/core/issues/18318
+     * 
+     * The servlet context is too dangerous to expose to velocity and allows for arbitrary 
+     * code execution
+     * 
+     * @throws Exception
+     */
+    
+    @Test
+    public void test_servlet_context_is_not_available_in_velocity() throws Exception {
+
+        final HttpServletRequest request = new MockSessionRequest(
+                        new MockAttributeRequest(new MockHttpRequest("localhost", "/").request()).request()).request();
+
+        final HttpServletResponse response = new MockHttpResponse().response();
+
+
+        ChainedContext context = (ChainedContext) VelocityUtil.getWebContext(request, response);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'context' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5378571)